src/modules/voxelDensity/process.cpp

```diff
@@ -0,0 +1,104 @@
+#include "analyser/dataExporter.h"
+#include "analyser/dataOperator1D.h"
+#include "main/dissolve.h"
+#include "math/gaussFit.h"
+#include "module/context.h"
+#include "voxelDensity.h"
+#include <cmath>
+
+void VoxelDensityModule::addValue(Vec3<double> coords, double value, Array3D<double> &array)
+{
+    auto t = std::make_tuple((int)std::floor(coords.x * nAxisVoxels_), (int)std::floor(coords.y * nAxisVoxels_),
+                             (int)std::floor(coords.z * nAxisVoxels_));
+    array[t] += value;
+}
+
+Vec3<double> VoxelDensityModule::foldedCoordinates(Atom &atom, const Box *unitCell)
+{
+    atom.set(unitCell->foldFrac(atom.r()));
+    auto x = atom.x(), y = atom.y(), z = atom.z();
+    return Vec3<double>(x, y, z);
```

Comment:
This is modifying the actual coordinates of the `Atom` and, while it's only a fold operation, we should be as read-only as possible on a `Configuration`, especially for analysis-type modules which aren't expected to modify contents.

If you change the function signature to just take the non-const `Vec3<double> r` of the atom coordinates, then you can do:
```suggestion
    return unitCell->foldFrac(r);
```
You may even wish to remove the function altogether and just do it in place since it's only one line.